addon/index.ts

```diff
@@ -1,6 +1,7 @@
-/* global requirejs:false, require:false */
-function resolveInitializer(moduleName) {
-  var module = require(moduleName, null, null, true);
+import Engine from '@ember/engine';
+
+function resolveInitializer(moduleName: string) {
+  var module = self.require(moduleName, null, null, true);
```

Comment:
using of `self` is needed to stop intercept with built-in Node's `require` typings

Reply:
Can you swap this back and add `import require from 'require'` to the top of the file?